anvil/src/udev.rs

```diff
@@ -700,6 +719,62 @@ impl AnvilState<UdevData> {
         }
     }
 
+    fn vblank(&mut self, dev_id: DrmNode, crtc: crtc::Handle, metadata: &mut Option<DrmEventMetadata>) {
+        let device_backend = match self.backend_data.backends.get_mut(&dev_id) {
+            Some(backend) => backend,
+            None => {
+                error!(self.log, "Trying to render on non-existent backend {}", dev_id);
+                return;
+            }
+        };
+        let surfaces = device_backend.surfaces.borrow();
+        let surface = match surfaces.get(&crtc) {
+            Some(surface) => surface,
+            None => {
+                error!(self.log, "Trying to render on non-existent backend {}", dev_id);
+                return;
+            }
+        };
+
+        let output = match self
+            .space
+            .outputs()
+            .find(|o| {
+                o.user_data().get::<UdevOutputId>()
+                    == Some(&UdevOutputId {
+                        device_id: surface.borrow().device_id,
+                        crtc,
+                    })
+            })
+            .cloned()
+        {
+            Some(output) => output,
+            None => return,
+        };
+
+        let report = match surface.borrow_mut().output_render_report.take() {
+            Some(report) => report,
+            None => return,
+        };
+        std::mem::drop(surfaces);
+
+        let tp = metadata.as_ref().and_then(|metadata| match metadata.time {
+            smithay::backend::drm::DrmEventTime::Monotonic(tp) => Some(tp),
```

Comment:
This case should also set the `hw_clock` (and possibly even the `hw_completion`) flags of the protocol, no?

Reply:
yeah, I think both should be set in this case. I think the best way would be make the ***base*** flags a parameter of `send_frames` so that the backend can provide them. For drm that would be `vsync | hw_clock | hw_completion` if we get a monotonic metadata. If we have no metadata or not a monotonic time we could fall back to `clock_gettime` and only send `vsync`. Does that sound good?